state/protocol/badger/mutator_test.go

```diff
@@ -723,30 +780,36 @@ func TestExtendReceiptsLatestSealed(t *testing.T) {
 		err = state.Mutate().Extend(&block2)
 		require.Nil(t, err)
 
-		// create a seals for block2
-		seal2 := unittest.SealFixture(
-			unittest.SealWithBlockID(block2.ID()),
-		)
-
-		// create block3 containing a seal for block2
+		// create block3 with a receipt for block2
+		block2Receipt := unittest.ReceiptForBlockFixture(&block2)
 		block3 := unittest.BlockWithParentFixture(block2.Header)
-		block3.Payload.Guarantees = nil
 		block3.SetPayload(flow.Payload{
-			Seals: []*flow.Seal{seal2},
+			Receipts: []*flow.ExecutionReceipt{block2Receipt},
 		})
-		block3.Header.PayloadHash = block3.Payload.Hash()
 		err = state.Mutate().Extend(&block3)
 		require.Nil(t, err)
 
-		// insert a receipt for block 2, which is now the highest sealed block
-		receipt := unittest.ReceiptForBlockFixture(&block2)
+		// create a seals for block2
+		seal2 := unittest.SealFixture(unittest.SealFromResult(&block2Receipt.ExecutionResult))
 
-		// test that the receipt is rejected
+		// create block4 containing a seal for block2
 		block4 := unittest.BlockWithParentFixture(block3.Header)
-		block4.Payload.Guarantees = nil
-		block4.Payload.Receipts = append(block4.Payload.Receipts, receipt)
-		block4.Header.PayloadHash = block4.Payload.Hash()
+		block4.SetPayload(flow.Payload{
+			Seals: []*flow.Seal{seal2},
+		})
 		err = state.Mutate().Extend(&block4)
+		require.Nil(t, err)
+
+		// insert another receipt for block 2, which is now the highest sealed
+		// block
+		receipt := unittest.ReceiptForBlockFixture(&block2)
+
+		// test that the receipt is rejected
```

Comment:
Would you mind just adding a quick goDoc at the top of `TestExtendReceiptsLatestSealed`? Thanks